psydac/api/postprocessing.py

```diff
@@ -79,3 +88,759 @@ def get_grid_lines_2d(domain_h, V_h, *, refine=1):
+        else:
+            assert spaces_info['ndim'] == pdim
+            try:
+                patch_index = [spaces_info['patches'][i]['name']
+                               for i in range(len(spaces_info['patches']))].index(patch)
+            except ValueError:
+                patch_index = -1
+
+            if patch_index != -1:
+                assert all(spaces_info['patches'][patch_index]['scalar_spaces'][i]['name'] != scalar_space_name
+                           for i in range(len(spaces_info['patches'][patch_index]['scalar_spaces'])))
+
+                spaces_info['patches'][patch_index]['scalar_spaces'].append(new_space)
+            else:
+                spaces_info['patches'].append({'name': patch, 'scalar_spaces': [new_space]})
+        self._spaces.append(scalar_space)
+        self._spaces.append(name)
+
+        self._spaces_info = spaces_info
+
+        return new_space
+
+    def _add_vector_space(self, vector_space, name=None):
+        """Add a vector space to the scope of this instance of OutputManager.
+
+        Parameters
+        ----------
+        vector_space: psydac.fem.vector.VectorFemSpace
+            Vector FemSpace to add to the scope.
+        """
+
+        symbolic_space = vector_space.symbolic_space
+        dim = symbolic_space.domain.dim
+        patch_name = symbolic_space.domain.name
+        kind = symbolic_space.kind
+
+        scalar_spaces_info = []
+        for i, scalar_space in enumerate(vector_space.spaces):
+            sc_space_info = self._add_scalar_space(scalar_space,
+                                                   name=name + f'[{i}]',
+                                                   dim=dim,
+                                                   patch_name=patch_name,
+                                                   kind='UndefinedSpaceType()')
+            scalar_spaces_info.append(sc_space_info)
+
+        spaces_info = self._spaces_info
+
+        new_vector_space = {'name': name,
+                            'kind': self.space_types_to_str[str(kind)],
+                            'components': scalar_spaces_info,
+                            }
+
+        patch_index = [spaces_info['patches'][i]['name']
+                       for i in range(len(spaces_info['patches']))].index(patch_name)
+
+        try:
+            spaces_info['patches'][patch_index]['vector_spaces'].append(new_vector_space)
+        except KeyError:
+            spaces_info['patches'][patch_index].update({'vector_spaces': [new_vector_space]})
+
+        self._spaces_info = spaces_info
+        self._spaces.append(vector_space)
+        self._spaces.append(name)
+
+    def export_fields(self, **fields):
+        """
+        A function that exports the fields' coefficients to an HDF5 file. They are
+        saved under a snapshot/patch/space/field scheme or static/patch/space/field.
+
+        If the saving scheme is time dependent, unsets it to avoid writing twice to
+        the same snapshot.
```

Comment:
What do you mean?

Reply:
I see.

I do not recall our discussion in every detail, but I imagine that there are situations where the user may want to export many fields, and maybe they are not all available at the same location in the code. So in this case it could be convenient to call `export_fields` multiple times on the same snapshot...